src/storage/storage.py

```diff
@@ -53,8 +53,8 @@ def clean(self):
         raise NotImplementedError
 
 
-class MailboxStorage(Storage, collections.MutableMapping):    # pylint: disable=abstract-method
+class MailboxStorage(Storage, collections.MutableMapping):
     """Method for storing mails"""
-    def __init__(self):
-        # pylint: disable=super-init-not-called
+    # pylint: disable=abstract-method
+    def __init__(self):   # pylint: disable=super-init-not-called
```

Comment:
This `__init__` makes no sense.

Reply:
Yeah, Discussed with peter too. MailboxStorage is an abstract class. Need more things to be added in that

Reply:
So make it abstract. What a problem with using [ABCMeta](https://docs.python.org/2.7/library/abc.html?highlight=abcmeta#abc.ABCMeta)?

Reply:
Like this?
> class MailboxStorage(Storage, collections.MutableMapping):
>     """Method for storing mails"""
>     __metaclass__ = ABCMeta
> 
>     @abstractmethod
>     def __init__(self):
>         pass

Reply:
Like this?
class MailboxStorage(Storage, collections.MutableMapping):
     """Method for storing mails"""
    __metaclass__ = ABCMeta

    @abstractmethod
    def __init__(self):
        pass

Reply:
without `__init__`, just pass

Reply:
`__init__` cannot be an abstract method by it's nature

Reply:
I am doing this now, but still got warnings. 
```
class MailboxStorage(Storage, collections.MutableMapping):
    """Method for storing mails"""
    __metaclass__ = ABCMeta

    pass
```
https://imgur.com/a/HTlWMOu

Reply:
Because you are doing it in the wrong place

```python
class InventoryStorage(collections.MutableMapping):
    """Base class for inventory storage"""
    numberOfInventoryLookupsPerformed = 0

    @abstractmethod
    def __contains__(self, _):
        pass

    @abstractmethod
    def __iter__(self):
        pass

    @abstractmethod
    def __len__(self):
        pass

    @abstractmethod
    def by_type_and_tag(self, objectType, tag):
        """Return objects filtered by object type and tag"""
        pass

    @abstractmethod
    def unexpired_hashes_by_stream(self, stream):
        """Return unexpired inventory vectors filtered by stream"""
        pass

    @abstractmethod
    def flush(self):
        """Flush cache"""
        pass

    @abstractmethod
    def clean(self):
        """Free memory / perform garbage collection"""
        pass
```

Notice `collections.MutableMapping` is abstract class itself

Reply:
MailboxStorage class isn't used currently

Reply:
The functions that are defined on MutuableMapping are already abstract, like __len__ is decorated as abstract. So, why we decorate with abstractmethod here too? We can just override methods that are static? 
Can you elaborate me something more about this, please?

Reply:
In the current code nothing is decorated. Why I decorated `__iter__`  and `__len__` in this example? I don't remember - it was taken from unfinished refactoring branch.